### PR TITLE
chore: Remove Pages Router / Use App Router - `/workflows`

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4276
- Remove `/workflows` routes from Pages Router and render them from App router.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Navigate to `/workflows/` and `/workflows/[workflow]`

### CODE REVIEW

**Comment by hbjORbj** [comment]
DO NOT MERGE UNTIL EITHER (1) REVALIDATION LOGIC FOR SSR IS ADDED or (2) SSR IS REVERTED

File: apps/web/app/workflows/page.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  // let filteredList;
+  // try {
+  //   filteredList = await WorkflowRepository.getFilteredList({
+  //     userId: user?.id,
+  //     input: {
+  //       filters,
+  //     },
+  //   });
+  // } catch (err) {}
```
Comment (hbjORbj) [inline apps/web/app/workflows/page.tsx:31]: commenting out sever-side fetching for now for defensive migration

File: apps/web/app/workflows/[workflow]/page.tsx [hunk 2]
```diff
@@ -30,31 +30,33 @@ export const generateStaticParams = () => [];
...
+  //     userId: user?.id ?? null,
+  //     teamId: workflow?.team?.id,
+  //   });
+  // } catch (err) {}
+  // try {
+  //   verifiedNumbers = await WorkflowRepository.getVerifiedNumbers({
+  //     userId: user?.id ?? null,
+  //     teamId: workflow?.team?.id,
+  //   });
+  // } catch (err) {}
```
Comment (hbjORbj) [inline apps/web/app/workflows/[workflow]/page.tsx:54]: commenting out sever-side fetching for now for defensive migration

**Review by eunjae-lee — APPROVED** [review]
working well!

**Pull request merged**